### ISSUE TITLE
Creating a new project under MacOS 15.7 "error Framework build type is static library"

### ISSUE THREAD

**Issue by sdancer75** [facebook/react-native#38109]
### Description

Error creating a new project.

`npx react-native --verbose init AwesomeProject `

```
sdancer@192 react-native-projects % npx react-native --verbose init AwesomeProject
debug We couldn't find a package.json in your project. Are you sure you are running it inside a React Native project?
debug Failed to load configuration of your project. Only a subset of commands will be available.
debug Initializing new project
                                                          
               ######                ######               
             ###     ####        ####     ###             
            ##          ###    ###          ##            
            ##             ####             ##            
            ##             ####             ##            
            ##           ##    ##           ##            
            ##         ###      ###         ##            
             ##  ########################  ##             
          ######    ###            ###    ######          
      ###     ##    ##              ##    ##     ###      
   ###         ## ###      ####      ### ##         ###   
  ##           ####      ########      ####           ##  
 ##             ###     ##########     ###             ## 
  ##           ####      ########      ####           ##  
   ###         ## ###      ####      ### ##         ###   
      ###     ##    ##              ##    ##     ###      
          ######    ###            ###    ######          
             ##  ########################  ##             
            ##         ###      ###         ##            
            ##           ##    ##           ##            
            ##             ####             ##            
            ##             ####             ##            
            ##          ###    ###          ##            
             ###     ####        ####     ###             
               ######                ######               
                                                          

                  Welcome to React Native!                
                 Learn once, write anywhere               

debug Installing template from react-native@latest
Wrote to /private/var/folders/1f/47lz1c1x655bn0ts9hdb9vhw0000gn/T/rncli-init-template-DjVbVk/package.json:

{
  "name": "rncli-init-template-djvbvk",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}


npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0

added 699 packages, and audited 700 packages in 19s

41 packages are looking for funding
  run `npm fund` for details

57 moderate severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
debug Getting config from /var/folders/1f/47lz1c1x655bn0ts9hdb9vhw0000gn/T/rncli-init-template-DjVbVk/node_modules/react-native/template.config.js
debug Copying template from /var/folders/1f/47lz1c1x655bn0ts9hdb9vhw0000gn/T/rncli-init-template-DjVbVk/node_modules/react-native/template
debug Changing HelloWorld for AwesomeProject in template
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/tsconfig.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/tsconfig.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/package.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/package.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/metro.config.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/metro.config.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/jest.config.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/jest.config.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/_xcode.env
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/_xcode.env
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/ios/_xcode.env -> file:/Users/sdancer/react-native-projects/AwesomeProject/ios/.xcode.env
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/Podfile
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/Podfile
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorldTests/Info.plist
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorldTests/Info.plist
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorldTests/HelloWorldTests.m
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorldTests/HelloWorldTests.m
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorldTests/HelloWorldTests.m -> file:/Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorldTests/AwesomeProjectTests.m
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorldTests -> file:/Users/sdancer/react-native-projects/AwesomeProject/ios/AwesomeProjectTests
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld.xcodeproj/xcshareddata/xcschemes/HelloWorld.xcscheme
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld.xcodeproj/xcshareddata/xcschemes/HelloWorld.xcscheme
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld.xcodeproj/xcshareddata/xcschemes/HelloWorld.xcscheme -> file:/Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld.xcodeproj/xcshareddata/xcschemes/AwesomeProject.xcscheme
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld.xcodeproj/project.pbxproj
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld.xcodeproj/project.pbxproj
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld.xcodeproj -> file:/Users/sdancer/react-native-projects/AwesomeProject/ios/AwesomeProject.xcodeproj
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/main.m
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/main.m
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/LaunchScreen.storyboard
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/LaunchScreen.storyboard
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/Info.plist
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/Info.plist
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/Images.xcassets/Contents.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/Images.xcassets/Contents.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/Images.xcassets/AppIcon.appiconset/Contents.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/Images.xcassets/AppIcon.appiconset/Contents.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/AppDelegate.mm
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/AppDelegate.mm
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/AppDelegate.h
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld/AppDelegate.h
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/ios/HelloWorld -> file:/Users/sdancer/react-native-projects/AwesomeProject/ios/AwesomeProject
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/index.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/index.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/babel.config.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/babel.config.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/app.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/app.json
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/settings.gradle
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/settings.gradle
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradlew.bat
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradlew.bat
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradlew
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradlew
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradle.properties
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradle.properties
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradle/wrapper/gradle-wrapper.properties
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradle/wrapper/gradle-wrapper.properties
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradle/wrapper/gradle-wrapper.jar
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/gradle/wrapper/gradle-wrapper.jar
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/build.gradle
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/build.gradle
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/release/java/com/helloworld/ReactNativeFlipper.java
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/release/java/com/helloworld/ReactNativeFlipper.java
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/release/java/com/helloworld -> file:/Users/sdancer/react-native-projects/AwesomeProject/android/app/src/release/java/com/awesomeproject
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/values/styles.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/values/styles.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/values/strings.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/values/strings.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-mdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-mdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-hdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/mipmap-hdpi/ic_launcher.png
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/drawable/rn_edit_text_material.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/res/drawable/rn_edit_text_material.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/java/com/helloworld/MainApplication.java
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/java/com/helloworld/MainApplication.java
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/java/com/helloworld/MainActivity.java
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/java/com/helloworld/MainActivity.java
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/java/com/helloworld -> file:/Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/java/com/awesomeproject
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/AndroidManifest.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/main/AndroidManifest.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/debug/java/com/helloworld/ReactNativeFlipper.java
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/debug/java/com/helloworld/ReactNativeFlipper.java
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/debug/java/com/helloworld -> file:/Users/sdancer/react-native-projects/AwesomeProject/android/app/src/debug/java/com/awesomeproject
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/debug/AndroidManifest.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/src/debug/AndroidManifest.xml
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/proguard-rules.pro
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/proguard-rules.pro
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/debug.keystore
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/debug.keystore
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/build.gradle
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/android/app/build.gradle
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_watchmanconfig
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_watchmanconfig
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/_watchmanconfig -> file:/Users/sdancer/react-native-projects/AwesomeProject/.watchmanconfig
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_prettierrc.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_prettierrc.js
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/_prettierrc.js -> file:/Users/sdancer/react-native-projects/AwesomeProject/.prettierrc.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_gitignore
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_gitignore
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/_gitignore -> file:/Users/sdancer/react-native-projects/AwesomeProject/.gitignore
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_eslintrc.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_eslintrc.js
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/_eslintrc.js -> file:/Users/sdancer/react-native-projects/AwesomeProject/.eslintrc.js
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_bundle/config
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/_bundle/config
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/_bundle/config -> file:/Users/sdancer/react-native-projects/AwesomeProject/_bundle/config
debug Renaming /Users/sdancer/react-native-projects/AwesomeProject/_bundle -> file:/Users/sdancer/react-native-projects/AwesomeProject/.bundle
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/__tests__/App.test.tsx
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/__tests__/App.test.tsx
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/Gemfile
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/Gemfile
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/App.tsx
debug Replacing in /Users/sdancer/react-native-projects/AwesomeProject/App.tsx
npm WARN deprecated @types/metro-config@0.76.3: This is a stub types definition. metro-config provides its own type definitions, so you do not need this installed.
npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0

added 1080 packages, and audited 1081 packages in 42s

122 packages are looking for funding
  run `npm fund` for details

83 moderate severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
error Framework build type is static library
[Codegen] Generating ./build/generated/ios/React-Codegen.podspec.json
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
[Codegen] Found FBReactNativeSpec
Fetching podspec for `RCT-Folly` from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`
[Codegen] Found rncore
Fetching podspec for `boost` from `../node_modules/react-native/third-party-podspecs/boost.podspec`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
Fetching podspec for `hermes-engine` from `../node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec`
[Hermes] Using the release tarball from Maven Central
[!] Couldn't determine repo type for URL: `https://cdn.cocoapods.org/`: SSL_connect returned=1 errno=0 peeraddr=172.67.74.167:443 state=error: certificate verify failed (unable to get local issuer certificate)
error Looks like your iOS environment is not properly set. Please go to https://reactnative.dev/docs/environment-setup?os=macos&platform=android and follow the React Native CLI QuickStart guide for macOS and iOS.

```

### React Native Version

0.72.0

### Output of `npx react-native info`

```
sdancer@192 react-native-projects % npx react-native info
error: unknown command 'info'
(Did you mean init?)
```

### Steps to reproduce

MacOS 10.15.7 (Catalina)
ruby version 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-darwin19]
pod version 1.12.1
watchman version 20230430.125247.0
node version 18.16.1
xcode 12.4

### Snack, code example, screenshot, or link to a repository

`npx react-native --verbose init AwesomeProject `

**Comment by cipolleschi** [comment]
Hi there, thanks for submitting the issue.
I'm a bit puzzled as that macOS version does not exists... :/ 
Ventura is 13.4
Sonoma, announced at WWDC this year, is still in Beta.

Were you meaning iOS, maybe? 

Also, the version of Xcode you should use is 14.3...

And Cocoapods is explaining what the error is:

```
[!] Couldn't determine repo type for URL: `https://cdn.cocoapods.org/`: SSL_connect returned=1 errno=0 peeraddr=172.67.74.167:443 state=error: certificate verify failed (unable to get local issuer certificate)
error Looks like your iOS environment is not properly set. Please go to https://reactnative.dev/docs/environment-setup?os=macos&platform=android and follow the React Native CLI QuickStart guide for macOS and iOS.
```

I think your environment is not properly set.

Please, follow the guide on reactnative.dev first and let us know if the issue is still happening.

**Comment by sdancer75** [comment]
I am sorry for the inconvenience, I corrected the MacOS version, the correct is 10.15.7 (Catalina)
Xcode = 12.4

I addressed the problem by re-installing Ruby version 2.7.5 and Cocoapods with Gem. It took me 3 days to find the correct version.... Coming from Windows OS I wrongly supposed that newer versions are better than older. Under MacOS this is not always true. MacOS is a problem itself!!

So, I will now close the issue.
